### PR TITLE
refactor!: porting frontend code to using react

### LINKED ISSUE #13: consider moving to frontend framework
Currently we're using express for serving, but don't have much structure around our frontend JavaScript/serving static assets in general.

Is there a lightweight framework we could use for this? The benefits I see:

* better separation between model/view.
* less likely to have a security hiccup if we use a framework that encourages good practices.
* potentially easier to make style consistent.

### PR DESCRIPTION
**Pull request by bcoe**
Someone recently asked for a new feature to Wombat Dressing Room (_the ability to lock down publication to specific npm tags_), and it motivated me to first dust off the codebase and migrate to using a front-end framework.

1. partially as a learning experience.
2. in an effort to make sure our front-end is well tested.

### Help page

Screenshot 2021-12-23 4 43 24 PM

### Login page

Screenshot 2021-12-23 4 42 53 PM

### Token management widget

Screenshot 2021-12-23 4 42 13 PM

### Token creation page

Screenshot 2021-12-23 5 08 55 PM

## Here's the state of frontend testing

Screenshot 2021-12-23 5 05 55 PM

tldr; we now ue React and have pretty good test coverage of the components.

Fixes #13

### CODE REVIEW

File: packages/server/src/server.ts [hunk 1]
```diff
@@ -235,35 +228,70 @@ const redirectToLoginServer = (req: express.Request, res: express.Response) => {
   return false;
 };
 
+/*
+ * If a session exists, return currently logged in account information.
+ */
 app.get(
```
Comment (ddelgrosso1) [inline packages/server/src/server.ts:234]: Just sort of a general comment, perhaps it would be nice to split all these routes out to a different file(s) and export the router from there as to streamline the server file a bit.

File: packages/app/src/manage-tokens.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      <CssBaseline />
+      <Nav manage={true} />
+      <Container maxWidth="md">
+        <h2>Manage Tokens</h2>
+        <TableContainer component={Paper}>
+          <Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
+            <TableHead>
+              <TableRow>
+                <TableCell>prefix</TableCell>
+                <TableCell align="right">package</TableCell>
```
Comment (ddelgrosso1) [inline packages/app/src/manage-tokens.tsx:73]: It looks like `align="right"` is repeated for most of these TableCell elements. Perhaps might make sense to break out a CSS file and apply the style at the element?

Comment (bcoe) [reply packages/app/src/manage-tokens.tsx:73]: It ended up looking better removing the `align` entirely.

File: packages/server/src/server.ts (hunk 1 above)
Comment (bcoe) [reply packages/server/src/server.ts:253]: I agree, I created this [tracking bug](https://github.com/GoogleCloudPlatform/wombat-dressing-room/issues/125) I would like to improve coverage in `server.ts` before we tear the file apart too much.

File: packages/app/src/nav.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      <AppBar position="static" color="primary">
+        <Toolbar>
+          <Tooltip title="Getting started" arrow>
+            <IconButton
+              data-testid="nav-help"
+              size="large"
+              aria-label="help"
+              aria-controls="menu-appbar"
+              color="inherit"
+              onClick={() => {
```
Comment (ddelgrosso1) [inline packages/app/src/nav.tsx:68]: Sort of nit picky but there is a mix of inline handlers and handler functions in this file, perhaps standardize on one.

Comment (bcoe) [reply packages/app/src/nav.tsx:68]: I refactored the JSX so that no multiline handlers are used. I still use closures in two places:

1. Adding handlers to a table, because this makes it easy to pass the current row to the handler:


```jsx
                    <Button variant="outlined" startIcon={<DeleteForever />} disabled={deleting} onClick={() => {
                      deleteToken(row)
                    }}>
```

2. I still think if it can be a one liner, it's nice to use a closure, so I kept this behavior:

```jsx
              onClick={() => {
                navigate('/_/help');
              }}
```
